packages/jest-snapshot/src/State.js

```diff
@@ -158,7 +158,7 @@ class SnapshotState {
       if (!pass) {
         this.unmatched++;
         return {
-          actual: receivedSerialized,
+          actual: receivedSerialized.replace(/\\"/g, '"'),
           count,
           expected,
```

Comment:
So, `expected` here is a serialized version. The way you unserialize a snapshot is by doing `require('snapshot.js')` which basically evals it.